### ISSUE TITLE
Accessing properties, weaving issue?

### ISSUE THREAD

**Issue by LeRondPoint** [realm/realm-dotnet#1418]
## Goals
On iOS: I'm trying to get the value of a property (file path) to load a picture.  depending on how I access the property, it's value is ok or null.  Even accessing the same object but after saving it in another variable yields differents results.  What am I missing here?

## Expected Results
Just getting my property value would be cool!

## Actual Results
See screenshot

## Steps to Reproduce
Launch the app, setup a picture by tapping the Toptal circled logo then fill form.  When going back to the login screen you should see your user and it'S picture.  Restart app, can't get the PicturePath property anymore so you won't see the user picture but using Bindings/Reflection to get properties works (like the name or below screenshot)...

![image](https://cloud.githubusercontent.com/assets/15251145/26516774/e20fe292-4259-11e7-8734-d66be33074c2.png)

## Code Sample
Use this project : 
https://github.com/LeRondPoint/FormsSkiaBikeTracker/tree/RealmPropertiesBug

## Version of Realm and Tooling
On version 1.3 reflection seems to return my property valuess.  On 1.4, I can't get a damn thing...  Feel free to update Nuget in branch to view that behavior.

**Comment by nirinchev** [comment]
I'm still struggling to run the project - in the simulator, the picture callback is never invoked. In any case, do you think it's possible that it's just a visualization bug in VS watch window and that it doesn't accurately represent the state of the object? My skepticism is based on the fact that accessing properties is fairly basic functionality that is extensively covered by tests and no other users have reported running into such a problem.

**Comment by LeRondPoint** [comment]
It's possible but I have the same symptoms as you.  Many properties don't hit my breakpoints and it's always when there is a Realm property involved.  My feeling is that there might be a weaving issue involved but I'm very surprised that such a basic feature is giving me issues.  As you said, this would have come up by now if it was widespread...  Could there be a conflict with the PropertyChanged.Fody package?

It is also possible that it is a VS debugger issue, again related to the weaving as it can be a side effect., but the UI seems to reflect the same behavior.  Anyways, my issue can also be related to the fact that I store the full file path, which I will fix once I resurrect my dying PC...

Or could it be a threading issue?  I remember reading that Realm instances are thread bound, so what happens when an object instance is propagated?

**Comment by nirinchev** [comment]
They are thread bound indeed, but based on your screenshot, it doesn't look like there's a threading issue (otherwise, you would be having problems with inspecting `view.Athlete`). I'll try to get an Android device tomorrow to try and replicate the issue, but my bet would be some peculiarity of VS rather than an actual bug. I would suggest you use `Console.WriteLine` and see if the value is indeed `null` in the log.

**Comment by LeRondPoint** [comment]
I'm unsure of the state of the sample on Android as I haven't tested it yet.  Normally it should work as it's all cross platform code but you might expect issues with the list display.

**Comment by LeRondPoint** [comment]
I've been testing on iOS 9, it's probably the info.plist entry missing for the camera... sorry

**Comment by nirinchev** [comment]
I managed to run it on iOS and I did observe the bug that you describe - the inspector shows properties as null incorrectly. I'm not sure what causes this behavior, but it looks like a bug in the inspector rather than Realm itself, because if I modify your code a little:

```csharp
string propPath = athlete.PicturePath;
string propReflectPath = pi.GetValue(athlete) as string;

if (propPath != propReflectPath)
{
    throw new Exception("propPath != propReflectPath");
}
```

The exception is never thrown, which means that your application correctly manages to obtain the values from a RealmObject's properties in both ways.

As for why the images are not visible, I'm not sure - I'm not familiar with the way images are data-bound and transformed from strings into streams, so there may be a bug there.
